File: hdu/绝对值.cpp
#include <iostream>
#include <vector>
#include <cstdlib> 

using i64 = long long; 

int main() {
    int n;
    std::cin >> n;

    i64 count = 0;
    std::vector<i64> a_pfx = {0}; 
    std::vector<i64> a_ppfx = {0}; 

    for (int i = 0; i < n; i++) {
        i64 val;
        std::cin >> val;

        a_pfx.push_back(a_pfx.back() + std::abs(val));
        a_ppfx.push_back(a_ppfx.back() + a_pfx.back());
    }
    std::cout << a_ppfx.back() << std::endl;

    return 0;
}

File: hdu/绝对值.py
# n = int(input())

# count = 0
# max_ = 0
# min_ = float("+inf")
# last = 0
# for i in map(int, input().split()):
#     last = last + abs(i)
#     max_ = max(max_, last)
#     min_ = min(min_, last)

#     count += max_ - min_
# print(count)
n = int(input())

count = 0
a_pfx = [0]
a_ppfx = [0]
for i in map(int, input().split()):
    a_pfx.append(a_pfx[-1] + abs(i))
    a_ppfx.append(a_ppfx[-1] + a_pfx[-1])
print(a_ppfx[-1])

